### DISCUSSION TITLE
"Shiny-Shadows" in nif model

### DISCUSSION THREAD

**Discussion by SonicMagna** [BadDogSkyrim/PyNifly#151] [Q&A] [answered]
I'm still learning so this might be 100% user error, but I can not figure this out.
Blender 3.5.1 with PyNifly V9.10
NifSkope 2.0 Dev 8, Fallout 4

The symptom I am trying to solve is my models (.nif) that I am making have a "shiny-shadow" at the far end.
I have tried every combo I can think of to get rid of this.

It is very easily repeatable. Open Blender, and export the cube model using Nifly set for Fallout 4.
You can see it in Nifskope as soon as you add a texture. And of course it shows up in game also.

Nifly will create the nif with a BSLightingShaderProperty already inserted - even if the Blender model has no lights/textures/etc. I have tried deleting that from the nif (Nifskope), as well as adding new ones, copying branches from other models, changing every number and setting I can see in both nifskope and blender... nothing works. It's like there is a setting hard-baked in somewhere.
Thanks for this work :)
![shinyShadow](https://github.com/BadDogSkyrim/PyNifly/assets/52136729/018303dc-b09f-479e-aea3-a105b50c3c1b)

**Comment by SonicMagna** [comment]
This is most likely lack of (my) user knowledge. I determined the Specular setting in Blender was the culprit. But whatever I set in Blender, I can not change in Nifskope in the BSLightingShaderProperty. I did not think it worked that way, but apparently it does.
![specularSetting](https://github.com/BadDogSkyrim/PyNifly/assets/52136729/a9aa9e4e-1d42-401c-958e-06c826ebfc59)

**Comment by BadDogSkyrim** [comment, marked as answer]
That's cuz FO4 doesn't actually use the LightingShaderProperty. Surprise. It takes the name of the LightingShaderProperty node and uses it as a relative path to get to a materials file and that's what it uses in game. I think nifskope does the same thing. I've had it on a vague sort of wish list to handle them properly in pyNifly too. 

Specular generally doesn't get set through the node or the materials file. It's generally set with a texture, at least the meshes I've worked with. Check the wiki, I think it's described there. For your issue, check specular strength and rim lighting. Copy values from the same type of object as you are creating.

**Reply by SonicMagna** [reply]
Ah. That sorted out my confusion. TYVM for the response and plug-in!

**Comment by BadDogSkyrim** [comment]
I just enabled github Discussions, so I'm converting this to a discussion over there. We can keep talking about it and even bring it back as an issue, if there turns out to be a real problem.
